dplyr not compiling on Windows 10

I'm getting a compile error when installing dplyr from GitHub.  Thanks, Bob

---


```r
Restarting R session...

> library("devtools")
> devtools::install_github("tidyverse/dplyr")
Downloading GitHub repo tidyverse/dplyr@master
from URL https://api.github.com/repos/tidyverse/dplyr/zipball/master
Installing dplyr
Downloading GitHub repo hadley/rlang@ac7498c9
from URL https://api.github.com/repos/hadley/rlang/zipball/ac7498c9
Installing rlang
"C:/PROGRA~1/R/R-33~1.3/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL  \
  "C:/Users/muenchen/AppData/Local/Temp/Rtmp0IyXF8/devtoolsdf87d535405/hadley-rlang-ac7498c"  \
  --library="C:/Program Files/R/R-3.3.3/library" --install-tests 

* installing *source* package 'rlang' ...
** libs
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c attrs.c -o attrs.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c capture.c -o capture.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c eval.c -o eval.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c export.c -o export.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c init.c -o init.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c internals.c -o internals.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c pairlist.c -o pairlist.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c replace-na.c -o replace-na.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c splice.c -o splice.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c symbol.c -o symbol.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c unquote.c -o unquote.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG     -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c utils.c -o utils.o
c:/Rtools/mingw_64/bin/gcc -shared -s -static-libgcc -o rlang.dll tmp.def attrs.o capture.o eval.o export.o init.o internals.o pairlist.o replace-na.o splice.o symbol.o unquote.o utils.o -Ld:/Compiler/gcc-4.9.3/local330/lib/x64 -Ld:/Compiler/gcc-4.9.3/local330/lib -LC:/PROGRA~1/R/R-33~1.3/bin/x64 -lR
installing to C:/Program Files/R/R-3.3.3/library/rlang/libs/x64
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (rlang)
"C:/PROGRA~1/R/R-33~1.3/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL  \
  "C:/Users/muenchen/AppData/Local/Temp/Rtmp0IyXF8/devtoolsdf841cd3a04/tidyverse-dplyr-8ba7bc9"  \
  --library="C:/Program Files/R/R-3.3.3/library" --install-tests 

* installing *source* package 'dplyr' ...
** libs
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c RcppExports.cpp -o RcppExports.o
In file included from C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/r/headers.h:50:0,
                 from C:/Program Files/R/R-3.3.3/library/Rcpp/include/RcppCommon.h:29,
                 from C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp.h:27,
                 from ../inst/include/dplyr/main.h:4,
                 from dplyr_types.h:1,
                 from RcppExports.cpp:4:
RcppExports.cpp: In function 'SEXPREC* dplyr_get_date_classes_try()':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9: warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;                                                                   \
         ^
RcppExports.cpp:256:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10: warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;                                                    \
          ^
RcppExports.cpp:256:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
RcppExports.cpp: In function 'SEXPREC* dplyr_get_time_classes_try()':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9: warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;                                                                   \
         ^
RcppExports.cpp:285:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10: warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;                                                    \
          ^
RcppExports.cpp:285:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
RcppExports.cpp: In function 'SEXPREC* dplyr_build_index_cpp_try(SEXP)':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9: warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;                                                                   \
         ^
RcppExports.cpp:314:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10: warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;                                                    \
          ^
RcppExports.cpp:314:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c address.cpp -o address.o
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c api.cpp -o api.o
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c arrange.cpp -o arrange.o
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c between.cpp -o between.o
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING   -I"C:/Program Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program Files/R/R-3.3.3/library/BH/include" -I"C:/Program Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program Files/R/R-3.3.3/library/plogr/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c bind.cpp -o bind.o
In file included from C:/Program Files/R/R-3.3.3/library/Rcpp/include/RcppCommon.h:179:0,
                 from C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp.h:27,
                 from pch.h:1,
                 from bind.cpp:1:
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h: In instantiation of 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown_iterable(const T&, Rcpp::traits::false_type) [with T = long long unsigned int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h:732:98:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown(const T&, Rcpp::traits::false_type) [with T = long long unsigned int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h:770:96:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_eigen(const T&, Rcpp::traits::false_type) [with T = long long unsigned int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h:787:80:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch_unknown_importable(const T&, Rcpp::traits::false_type) [with T = long long unsigned int; SEXP = SEXPREC*; Rcpp::traits::false_type = Rcpp::traits::integral_constant<bool, false>]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h:807:99:   required from 'SEXPREC* Rcpp::internal::wrap_dispatch(const T&, Rcpp::traits::wrap_type_unknown_tag) [with T = long long unsigned int; SEXP = SEXPREC*]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap_end.h:30:110:   required from 'SEXPREC* Rcpp::wrap(const T&) [with T = long long unsigned int; SEXP = SEXPREC*]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/grow.h:49:46:   required from 'SEXPREC* Rcpp::internal::grow__dispatch(Rcpp::traits::true_type, const T&, SEXP) [with T = Rcpp::traits::named_object<long long unsigned int>; SEXP = SEXPREC*; Rcpp::traits::true_type = Rcpp::traits::integral_constant<bool, true>]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/grow.h:65:86:   required from 'SEXPREC* Rcpp::grow(const T&, SEXP) [with T = Rcpp::traits::named_object<long long unsigned int>; SEXP = SEXPREC*]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/generated/grow__pairlist.h:50:60:   required from 'SEXPREC* Rcpp::pairlist(const T1&, const T2&, const T3&, const T4&) [with T1 = SEXPREC*; T2 = SEXPREC*; T3 = Rcpp::traits::named_object<long long unsigned int>; T4 = Rcpp::traits::named_object<long long unsigned int>; SEXP = SEXPREC*]'
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/generated/Function__operator.h:50:74:   required from 'SEXPREC* Rcpp::Function_Impl<StoragePolicy>::operator()(const T1&, const T2&, const T3&, const T4&) const [with T1 = SEXPREC*; T2 = SEXPREC*; T3 = Rcpp::traits::named_object<long long unsigned int>; T4 = Rcpp::traits::named_object<long long unsigned int>; StoragePolicy = Rcpp::PreserveStorage; SEXP = SEXPREC*]'
../inst/include/dplyr/bad.h:44:48:   required from 'void dplyr::bad_pos_arg(int, C1, C2, C3) [with C1 = const char*; C2 = Rcpp::traits::named_object<long long unsigned int>; C3 = Rcpp::traits::named_object<long long unsigned int>]'
bind.cpp:86:84:   required from here
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/internal/wrap.h:525:11: error: invalid conversion from 'long long unsigned int' to 'SEXP' [-fpermissive]
  SEXP x = object ;
           ^
make: *** [bind.o] Error 1
Warning: running command 'make -f "Makevars.win" -f "C:/PROGRA~1/R/R-33~1.3/etc/x64/Makeconf" -f "C:/PROGRA~1/R/R-33~1.3/share/make/winshlib.mk" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="dplyr.dll" WIN=64 TCLBIN=64 OBJECTS="RcppExports.o address.o api.o arrange.o between.o bind.o combine_variables.o distinct.o filter.o group_by.o group_indices.o hybrid.o hybrid_count.o hybrid_debug.o hybrid_in.o hybrid_minmax.o hybrid_nth.o hybrid_offset.o hybrid_simple.o hybrid_window.o init.o init_register.o join.o join_exports.o mutate.o rlang-export.o select.o set.o slice.o summarise.o test.o utils.o window.o"' had status 2
ERROR: compilation failed for package 'dplyr'
* removing 'C:/Program Files/R/R-3.3.3/library/dplyr'
* restoring previous 'C:/Program Files/R/R-3.3.3/library/dplyr'
Error: Command failed (1)
> sessionInfo()
R version 3.3.3 (2017-03-06)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] devtools_1.12.0

loaded via a namespace (and not attached):
[1] httr_1.2.1    R6_2.2.0      tools_3.3.3   withr_1.0.2   curl_2.3      memoise_1.0.0 knitr_1.15.1 
[8] git2r_0.18.0  digest_0.6.12
```

Hi Lionel, there's still a problem. Is it working for you? -Bob

> devtools::install_github("tidyverse/dplyr")
Downloading GitHub repo tidyverse/dplyr@master
from URL https://api.github.com/repos/tidyverse/dplyr/zipball/master
Installing dplyr
Downloading GitHub repo hadley/rlang@ac7498c9
from URL https://api.github.com/repos/hadley/rlang/zipball/ac7498c9
Installing rlang
"C:/PROGRA~1/R/R-33~1.3/bin/x64/R" --no-site-file --no-environ --no-save
--no-restore --quiet CMD  \
  INSTALL  \

"C:/Users/muenchen/AppData/Local/Temp/Rtmp0IyXF8/devtoolsdf869ed3f36/hadley-rlang-ac7498c"
 \
  --library="C:/Program Files/R/R-3.3.3/library" --install-tests

* installing *source* package 'rlang' ...
** libs
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c attrs.c -o attrs.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c capture.c -o capture.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c eval.c -o eval.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c export.c -o export.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c init.c -o init.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c internals.c -o internals.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c pairlist.c -o pairlist.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c replace-na.c -o replace-na.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c splice.c -o splice.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c symbol.c -o symbol.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c unquote.c -o unquote.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99
-mtune=core2 -c utils.c -o utils.o
c:/Rtools/mingw_64/bin/gcc -shared -s -static-libgcc -o rlang.dll tmp.def
attrs.o capture.o eval.o export.o init.o internals.o pairlist.o
replace-na.o splice.o symbol.o unquote.o utils.o
-Ld:/Compiler/gcc-4.9.3/local330/lib/x64
-Ld:/Compiler/gcc-4.9.3/local330/lib -LC:/PROGRA~1/R/R-33~1.3/bin/x64 -lR
installing to C:/Program Files/R/R-3.3.3/library/rlang/libs/x64
** R
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (rlang)
"C:/PROGRA~1/R/R-33~1.3/bin/x64/R" --no-site-file --no-environ --no-save
--no-restore --quiet CMD  \
  INSTALL  \

"C:/Users/muenchen/AppData/Local/Temp/Rtmp0IyXF8/devtoolsdf83bf531d9/tidyverse-dplyr-3d914be"
 \
  --library="C:/Program Files/R/R-3.3.3/library" --install-tests

* installing *source* package 'dplyr' ...
** libs
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.3/include" -DNDEBUG
-I../inst/include -DCOMPILING_DPLYR -DRCPP_USING_UTF8_ERROR_STRING
-DBOOST_NO_INTEGRAL_INT64_T -DBOOST_NO_LONG_LONG   -I"C:/Program
Files/R/R-3.3.3/library/Rcpp/include" -I"C:/Program
Files/R/R-3.3.3/library/BH/include" -I"C:/Program
Files/R/R-3.3.3/library/bindrcpp/include" -I"C:/Program
Files/R/R-3.3.3/library/plogr/include"
-I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c
RcppExports.cpp -o RcppExports.o
In file included from C:/Program
Files/R/R-3.3.3/library/Rcpp/include/Rcpp/r/headers.h:50:0,
                 from C:/Program
Files/R/R-3.3.3/library/Rcpp/include/RcppCommon.h:29,
                 from C:/Program
Files/R/R-3.3.3/library/Rcpp/include/Rcpp.h:27,
                 from ../inst/include/dplyr/main.h:4,
                 from dplyr_types.h:1,
                 from RcppExports.cpp:4:
RcppExports.cpp: In function 'SEXPREC* dplyr_get_date_classes_try()':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9:
warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;
                      \
         ^
RcppExports.cpp:256:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10:
warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;
                       \
          ^
RcppExports.cpp:256:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
RcppExports.cpp: In function 'SEXPREC* dplyr_get_time_classes_try()':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9:
warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;
                      \
         ^
RcppExports.cpp:285:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10:
warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;
                       \
          ^
RcppExports.cpp:285:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
RcppExports.cpp: In function 'SEXPREC* dplyr_build_index_cpp_try(SEXP)':
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:31:9:
warning: unused variable 'rcpp_output_type' [-Wunused-variable]
     int rcpp_output_type = 0 ;
                      \
         ^
RcppExports.cpp:314:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
C:/Program Files/R/R-3.3.3/library/Rcpp/include/Rcpp/macros/macros.h:32:10:
warning: unused variable 'rcpp_output_condition' [-Wunused-variable]
     SEXP rcpp_output_condition = R_NilValue ;
                       \
          ^
RcppExports.cpp:314:1: note: in expansion of macro 'BEGIN_RCPP'
 BEGIN_RCPP
 ^
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
error: no matching function for call to 'hash_value(const long long
unsigned int&)'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note: candidates are:
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6:0,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:258:57:
note: template<class T> typename boost::hash_detail::basic_numbers<T>::type
boost::hash_value(T)
     typename boost::hash_detail::basic_numbers<T>::type hash_value(T v)
                                                         ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:258:57:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp: In
substitution of 'template<class T> typename
boost::hash_detail::basic_numbers<T>::type boost::hash_value(T) [with T =
long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
  required from 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:258:57:
error: no type named 'type' in 'struct
boost::hash_detail::basic_numbers<long long unsigned int>'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:264:56:
note: template<class T> typename boost::hash_detail::long_numbers<T>::type
boost::hash_value(T)
     typename boost::hash_detail::long_numbers<T>::type hash_value(T v)
                                                        ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:264:56:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp: In
substitution of 'template<class T> typename
boost::hash_detail::long_numbers<T>::type boost::hash_value(T) [with T =
long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
  required from 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:264:56:
error: no type named 'type' in 'struct
boost::hash_detail::long_numbers<long long unsigned int>'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:270:57:
note: template<class T> typename boost::hash_detail::ulong_numbers<T>::type
boost::hash_value(T)
     typename boost::hash_detail::ulong_numbers<T>::type hash_value(T v)
                                                         ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:270:57:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp: In
substitution of 'template<class T> typename
boost::hash_detail::ulong_numbers<T>::type boost::hash_value(T) [with T =
long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
  required from 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:270:57:
error: no type named 'type' in 'struct
boost::hash_detail::ulong_numbers<long long unsigned int>'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:277:9:
note: template<class T> typename boost::enable_if<boost::is_enum<T>, long
long unsigned int>::type boost::hash_value(T)
         hash_value(T v)
         ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:277:9:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp: In
substitution of 'template<class T> typename
boost::enable_if<boost::is_enum<T>, long long unsigned int>::type
boost::hash_value(T) [with T = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
  required from 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:277:9:
error: no type named 'type' in 'struct boost::enable_if<boost::is_enum<long
long unsigned int>, long long unsigned int>'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:284:36:
note: template<class T> std::size_t boost::hash_value(T* const&)
     template <class T> std::size_t hash_value(T* const& v)
                                    ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:284:36:
note:   template argument deduction/substitution failed:
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'T* const' and 'const long long unsigned int'
             return hash_value(val);
                                  ^
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6:0,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:372:24:
note: template<class T, unsigned int N> std::size_t boost::hash_value(const
T (&)[N])
     inline std::size_t hash_value(const T (&x)[N])
                        ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:372:24:
note:   template argument deduction/substitution failed:
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const T [N]' and 'const long long unsigned int'
             return hash_value(val);
                                  ^
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6:0,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:378:24:
note: template<class T, unsigned int N> std::size_t boost::hash_value(T
(&)[N])
     inline std::size_t hash_value(T (&x)[N])
                        ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:378:24:
note:   template argument deduction/substitution failed:
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'T [N]' and 'const long long unsigned int'
             return hash_value(val);
                                  ^
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6:0,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:385:24:
note: template<class Ch, class A> std::size_t boost::hash_value(const
std::basic_string<Ch, std::char_traits<_CharT>, A>&)
     inline std::size_t hash_value(
                        ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:385:24:
note:   template argument deduction/substitution failed:
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::basic_string<Ch,
std::char_traits<_CharT>, A>' and 'const long long unsigned int'
             return hash_value(val);
                                  ^
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6:0,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:392:57:
note: template<class T> typename boost::hash_detail::float_numbers<T>::type
boost::hash_value(T)
     typename boost::hash_detail::float_numbers<T>::type hash_value(T v)
                                                         ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:392:57:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp: In
substitution of 'template<class T> typename
boost::hash_detail::float_numbers<T>::type boost::hash_value(T) [with T =
long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
  required from 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]'
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:392:57:
error: no type named 'type' in 'struct
boost::hash_detail::float_numbers<long long unsigned int>'
In file included from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:560:0,
                 from C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash.hpp:6,
                 from ../inst/include/tools/hash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4,
                 from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5,
                 from ../inst/include/dplyr/DataFrameVisitors.h:6,
                 from ../inst/include/dplyr/DataFrameVisitorsIndexMap.h:4,
                 from ../inst/include/dplyr/SubsetVectorVisitor.h:5,
                 from ../inst/include/dplyr/subset_visitor.h:4,
                 from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9,
                 from ../inst/include/dplyr/Result/GroupedSubset.h:6,
                 from ../inst/include/dplyr/GroupedDataFrame.h:7,
                 from dplyr_types.h:6,
                 from RcppExports.cpp:4:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp: In
instantiation of 'std::size_t boost::hash<T>::operator()(const T&) const
[with T = long long unsigned int; std::size_t = long long unsigned int]':
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/hash.hpp:315:68:
required from 'void boost::hash_combine(std::size_t&, const T&) [with T =
long long unsigned int; std::size_t = long long unsigned int]'
../inst/include/tools/hash.h:27:41:   required from here
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:67:17:
note: template<class A, class B> std::size_t boost::hash_value(const
std::pair<_T1, _T2>&)
     std::size_t hash_value(std::pair<A, B> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:67:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::pair<_T1, _T2>' and 'const long long
unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:76:17:
note: template<class T, class A> std::size_t boost::hash_value(const
std::vector<_Tp, _Alloc>&)
     std::size_t hash_value(std::vector<T, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:76:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::vector<_Tp, _Alloc>' and 'const long
long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:82:17:
note: template<class T, class A> std::size_t boost::hash_value(const
std::list<_Tp, _Alloc>&)
     std::size_t hash_value(std::list<T, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:82:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::list<_Tp, _Alloc>' and 'const long
long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:88:17:
note: template<class T, class A> std::size_t boost::hash_value(const
std::deque<_Tp, _Alloc>&)
     std::size_t hash_value(std::deque<T, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:88:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::deque<_Tp, _Alloc>' and 'const long
long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:94:17:
note: template<class K, class C, class A> std::size_t
boost::hash_value(const std::set<_Key, _Compare, _Alloc>&)
     std::size_t hash_value(std::set<K, C, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:94:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::set<_Key, _Compare, _Alloc>' and
'const long long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:100:17:
note: template<class K, class C, class A> std::size_t
boost::hash_value(const std::multiset<_Key, _Compare, _Alloc>&)
     std::size_t hash_value(std::multiset<K, C, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:100:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::multiset<_Key, _Compare, _Alloc>' and
'const long long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:106:17:
note: template<class K, class T, class C, class A> std::size_t
boost::hash_value(const std::map<_Key, _Tp, _Compare, _Alloc>&)
     std::size_t hash_value(std::map<K, T, C, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:106:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::map<_Key, _Tp, _Compare, _Alloc>' and
'const long long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:112:17:
note: template<class K, class T, class C, class A> std::size_t
boost::hash_value(const std::multimap<_Key, _Tp, _Compare, _Alloc>&)
     std::size_t hash_value(std::multimap<K, T, C, A> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:112:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::multimap<_Key, _Tp, _Compare, _Alloc>'
and 'const long long unsigned int'
             return hash_value(val);
                                  ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:118:17:
note: template<class T> std::size_t boost::hash_value(const
std::complex<_Tp>&)
     std::size_t hash_value(std::complex<T> const& v)
                 ^
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:118:17:
note:   template argument deduction/substitution failed:
C:/Program
Files/R/R-3.3.3/library/BH/include/boost/functional/hash/extensions.hpp:262:34:
note:   mismatched types 'const std::complex<_Tp>' and 'const long long
unsigned int'
             return hash_value(val);
                                  ^
make: *** [RcppExports.o] Error 1
Warning: running command 'make -f "Makevars.win" -f
"C:/PROGRA~1/R/R-33~1.3/etc/x64/Makeconf" -f
"C:/PROGRA~1/R/R-33~1.3/share/make/winshlib.mk"
SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)'
SHLIB="dplyr.dll" WIN=64 TCLBIN=64 OBJECTS="RcppExports.o address.o api.o
arrange.o between.o bind.o combine_variables.o distinct.o filter.o
group_by.o group_indices.o hybrid.o hybrid_count.o hybrid_debug.o
hybrid_in.o hybrid_minmax.o hybrid_nth.o hybrid_offset.o hybrid_simple.o
hybrid_window.o init.o init_register.o join.o join_exports.o mutate.o
rlang-export.o select.o set.o slice.o summarise.o test.o utils.o window.o"'
had status 2
ERROR: compilation failed for package 'dplyr'
* removing 'C:/Program Files/R/R-3.3.3/library/dplyr'
* restoring previous 'C:/Program Files/R/R-3.3.3/library/dplyr'
Error: Command failed (1)
>

On Wed, Apr 12, 2017 at 9:28 AM, Lionel Henry <notifications@github.com>
wrote:

> Could you try again please
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tidyverse/dplyr/issues/2638#issuecomment-293575734>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC_53PqGfVBnRcpJxzqHCfEE0pO5g3Jaks5rvNGFgaJpZM4M7ZBK>
> .
>



-- 

==================================================
  Bob Muenchen (pronounced MINchen)
  Accredited Professional Statistician™
  Manager, Research Computing Support, U of TN
  Voice: (865) 974-5230
  Email: muenchen@utk.edu
  Twitter: @BobMuenchen
  Web Site: http://r4stats.com
==================================================

--- a/inst/include/dplyr/main.h
+++ b/inst/include/dplyr/main.h
@@ -2,10 +2,11 @@
 #define dplyr_dplyr_main_H
 
 #include <Rcpp.h>
+#include <dplyr/workarounds.h>
 #include <dplyr/workarounds/static_assert.h>
+#include <dplyr/workarounds/xlen.h>
 #include <solaris/solaris.h>
 #include <dplyr/config.h>
-#include <dplyr/workarounds.h>
 
 #include <plogr.h>
 

--- a/inst/include/dplyr/workarounds/xlen.h
+++ b/inst/include/dplyr/workarounds/xlen.h
@@ -1,0 +1,18 @@
+#ifndef DPLYR_WORKAROUND_XLEN_H
+#define DPLYR_WORKAROUND_XLEN__H
+
+namespace Rcpp {
+
+template <>
+inline SEXP wrap(const R_xlen_t& x) {
+  if (x < -2147483647 || x > 2147483647) {
+    return Rf_ScalarReal(static_cast<double>(x));
+  }
+  else {
+    return Rf_ScalarInteger(static_cast<int>(x));
+  }
+}
+
+}
+
+#endif

--- a/src/bind.cpp
+++ b/src/bind.cpp
@@ -38,25 +38,25 @@ int df_rows_length(SEXP df) {
 }
 
 static
-size_t rows_length(SEXP x, bool rowwise) {
+R_xlen_t rows_length(SEXP x, bool rowwise) {
   if (TYPEOF(x) == VECSXP) {
     if (Rf_inherits(x, "data.frame"))
       return df_rows_length(x);
-    else if (Rf_length(x) > 0)
-      return Rf_length(VECTOR_ELT(x, 0));
+    else if (Rf_xlength(x) > 0)
+      return Rf_xlength(VECTOR_ELT(x, 0));
     else
       return 0;
   } else {
     if (rowwise)
       return 1;
     else
-      return Rf_length(x);
+      return Rf_xlength(x);
   }
 }
 static
-size_t cols_length(SEXP x) {
+R_xlen_t cols_length(SEXP x) {
   if (TYPEOF(x) == VECSXP)
-    return Rf_length(x);
+    return Rf_xlength(x);
   else
     return 1;
 }
@@ -80,7 +80,7 @@ void inner_vector_check(SEXP x, int nrows, int arg) {
 }
 
 static
-void rbind_vector_check(SEXP x, size_t nrows, int arg) {
+void rbind_vector_check(SEXP x, R_xlen_t nrows, int arg) {
   if (rows_length(x, true) != nrows) {
     bad_pos_arg(arg + 1, "size must be {expected_size}, not {actual_size}",
                 _["expected_size"] = rows_length(x, true), _["actual_size"] = nrows);
@@ -109,7 +109,7 @@ void rbind_vector_check(SEXP x, size_t nrows, int arg) {
               _["type"] = get_single_class(x));
 }
 static
-void cbind_vector_check(SEXP x, size_t nrows, SEXP contr, int arg) {
+void cbind_vector_check(SEXP x, R_xlen_t nrows, SEXP contr, int arg) {
   if (is_atomic(x) && !has_name_at(contr, arg))
     bad_pos_arg(arg + 1, "must have names");
   if (rows_length(x, false) != nrows) {
@@ -178,9 +178,9 @@ SEXP flatten_bindable(SEXP x) {
 
 List rbind__impl(List dots, SEXP id = R_NilValue) {
   int ndata = dots.size();
-  int n = 0;
+  R_xlen_t n = 0;
   std::vector<SEXP> chunks;
-  std::vector<int> df_nrows;
+  std::vector<R_xlen_t> df_nrows;
   std::vector<String> dots_names;
 
   chunks.reserve(ndata);
@@ -192,7 +192,7 @@ List rbind__impl(List dots, SEXP id = R_NilValue) {
     SEXP obj = dots[i];
     if (Rf_isNull(obj)) continue;
     chunks.push_back(obj);
-    size_t nrows = rows_length(chunks[k], true);
+    R_xlen_t nrows = rows_length(chunks[k], true);
     df_nrows.push_back(nrows);
     n += nrows;
     if (!Rf_isNull(id)) {
@@ -211,7 +211,7 @@ List rbind__impl(List dots, SEXP id = R_NilValue) {
     Rcpp::checkUserInterrupt();
 
     SEXP df = chunks[i];
-    int nrows = df_nrows[i];
+    R_xlen_t nrows = df_nrows[i];
     rbind_type_check(df, nrows, i);
 
     CharacterVector df_names = enc2native(vec_names(df));
@@ -230,7 +230,7 @@ List rbind__impl(List dots, SEXP id = R_NilValue) {
       String name = df_names[j];
 
       Collecter* coll = 0;
-      size_t index = 0;
+      R_xlen_t index = 0;
       for (; index < names.size(); index++) {
         if (name == names[index]) {
           coll = columns[index];
@@ -344,10 +344,10 @@ List cbind_all(List dots) {
     return DataFrame();
 
   SEXP first = dots[first_i];
-  const size_t nrows = rows_length(first, false);
+  const R_xlen_t nrows = rows_length(first, false);
   cbind_type_check(first, nrows, dots, 0);
 
-  size_t nv = cols_length(first);
+  R_xlen_t nv = cols_length(first);
 
   for (int i = first_i + 1; i < n_dots; i++) {
     SEXP current = dots[i];

--- a/src/bind.cpp
+++ b/src/bind.cpp
@@ -230,7 +230,7 @@ List rbind__impl(List dots, SEXP id = R_NilValue) {
       String name = df_names[j];
 
       Collecter* coll = 0;
-      R_xlen_t index = 0;
+      size_t index = 0;
       for (; index < names.size(); index++) {
         if (name == names[index]) {
           coll = columns[index];

